server/src/main/java/io/druid/server/coordination/ZkCoordinator.java

```diff
@@ -256,8 +256,14 @@ public void loadLocalCache()
       File file = segmentsToLoad[i];
       log.info("Loading segment cache file [%d/%d][%s].", i, segmentsToLoad.length, file);
       try {
-        DataSegment segment = jsonMapper.readValue(file, DataSegment.class);
-        if (serverManager.isSegmentCached(segment)) {
+        final DataSegment segment = jsonMapper.readValue(file, DataSegment.class);
+
+        if (!segment.getIdentifier().equals(file.getName())) {
+          log.makeAlert("Ignoring cache file for dataSource[%s].", segment.getDataSource())
```

Comment:
Does this have the ability to create a ton of alerts?

If this is going to continue on regardless, can the alert simply be that XXX segments were ignored?

Logging locally as WARN per segment could be OK.